sklearn/utils/_estimator_html_repr.py

```diff
@@ -381,10 +261,36 @@ def estimator_html_repr(estimator):
     html: str
         HTML representation of estimator.
     """
+    from sklearn.exceptions import NotFittedError
+    from sklearn.utils.validation import check_is_fitted
+
+    if not hasattr(estimator, "fit"):
+        # The estimator has no fit method, it's considered unfitted.
+        is_fitted = False
+        status_label = "<span>Estimator is not fitted</span>"
+        fitted_str = ""
+    else:
+        # The estimator has a proper fit method, so we can check if it's fitted.
+        try:
+            check_is_fitted(estimator)  # check if the estimator is fitted
+            is_fitted = True
+            status_label = "<span>Estimator is fitted</span>"
+            fitted_str = " fitted"  # `fitted_str` specifies the css class to use
```

Comment:
Is there a way to have this so that we don't need to have a leading white space? Important leading white space is the kind of thing that is easy to overlook/break by someone from the future. Maybe we can insert the white space at the location where `fitted_str` is used instead of here?

I think having an extra trailing white space in HTML attributes is OK/is legal HTML.

Reply:
The space was added to avoid failing some of the tests due to the formatting that is expected from the output, I didn't think about the issues, so that needs fixing.

Reply:
I reformatted both tests and html code to address this.